### DISCUSSION TITLE
Add Field (Column) Selection to the Terraform provider

### DISCUSSION THREAD

**Discussion by nataliekwong** [airbytehq/airbyte#41944] [Connector Ideas and Features]
Currently, fieldSelectionEnabled is not exposed to the Terraform provider.

Branched from this issue: https://github.com/airbytehq/airbyte/issues/35798

**Comment by kev-datams** [comment]
We would love to use this new capability 🙏

**Comment by jonseymour** [comment]
You can set it in the terraform provider now by adding selected_fields to each stream

For example:

```
        "selected_fields": [
          {
            "field_path": ["id"]
          },
          {
            "field_path": ["updated_at"]
          },
          {
            "field_path": ["email"]
          }
        ],
```

You do need to include the primary_key and cursor_fields in the list of selected fields.

The most glaring omission, however, is that neither the terraform provider nor the underlying SDK/API supports extracting the list of selected fields in a stream. This makes round-tripping impossible with the API alone and cumbersome otherwise,

See also: https://github.com/airbytehq/airbyte/discussions/45455

**Reply by JonsSpaghetti** [reply]
Hi @jonseymour , have you seen https://reference.airbyte.com/reference/getstreamproperties? I believe that allows you to pull the streams in the way you're describing. Not super well documented, but something we're hoping to socialize more as you're not the only person who has run into this.

**Comment by kev-datams** [comment]
@jonseymour it looks OK, could you retry ?
As explained [here](https://github.com/airbytehq/terraform-provider-airbyte/issues/124#issuecomment-2599043637), I successfully:
- defined some fields in a my stream Terraform configuration
- run `terraform apply` and saw impacted streams in the UI (+ trigger the sync in success)
- run `terraform plan` which displayed no changes 🙏
